### PR TITLE
cli: support `wasi_unstable` namespace

### PR DESCRIPTION
**Pull request by evacchi**
While playing with interesting examples to show off Wazero CLI,
I found that older binaries that still use the `wasi_unstable`
namespace are rejected.

This is a small refactoring to support running these binaries
as well.

For instance:

```bash
   curl -LO https://registry-cdn.wapm.io/contents/syrusakbary/cowsay/0.2.0/target/wasm32-wasi/release/cowsay.wasm
   wazero run cowsay.wasm Wazero is Awesome!
```

Signed-off-by: Edoardo Vacchi <evacchi@users.noreply.github.com>

### CODE REVIEW

File: cmd/wazero/wazero.go [hunk 1]
```diff
@@ -322,19 +331,28 @@ func validateMounts(mounts sliceFlag, stdErr logging.Writer, exit func(code int)
 	return
 }
 
-func detectImports(imports []api.FunctionDefinition) (needsWASI, needsGo bool) {
+const (
+	ModeDefault ImportMode = iota
+	ModeWasi
+	ModeWasiUnstable
+	ModeGo
```
Comment (mathetake) [inline cmd/wazero/wazero.go:338]: no need to export

File: cmd/wazero/wazero.go [hunk 2]
```diff
@@ -322,19 +331,28 @@ func validateMounts(mounts sliceFlag, stdErr logging.Writer, exit func(code int)
...
+)
+
+type ImportMode uint
```
Comment (mathetake) [inline cmd/wazero/wazero.go:341]: same!

**Review by codefromthecrypt — APPROVED** [review]
LGTM also with nits mentioned just for hygiene. Thanks!

**Comment by evacchi** [comment]
I guess I still haven't internalized capital letter => exported symbol 😅

**Pull request merged**